shaptools/hdb_connector/connectors/base_connector.py

```diff
@@ -34,6 +34,31 @@ class QueryError(BaseError):
     Error during query
     """
 
+class QueryResult(object):
+    """
+    Class to manage query results
+
+    Args:
+        cursor (cursor): sql cursor object
+    """
+
+    def __init__(self):
```

Comment:
I would check some format in the QueryResult class. I would do the next:

Add the needed parameters to the `__init__` method to have something like:
```
def __init__(self, records, metadata):
        self._logger = logging.getLogger(__name__)
        self.records = records
        self.metadata = metadata
```
This way we have the easiest way of creating and instance with info.
And them, if we want to gather data from a cursor directly add a class method. Something like:
```
@classmethod
def load_cursor(cls, cursor):
    """
    docstring...
    """
    instance = cls()
    instance.records = cursor.fetchall() # we should check if this can raise any exception...
    instance.metadata = cursor.description
    self._logger.info('query records: %s', instance.records)
    return instance
```
 With this, what we have is a really easy way to create an instance using a cursor as an input parameter.
This way we have the 2 options, it's really flexible.

You could do:
```
result = QueryResult(records, metadata)
# or
result = QueryResult.load_cursor(cursor)
```
And both will return the same object.